spec/lib/salus/scanners/slither_spec.rb

```diff
@@ -0,0 +1,153 @@
+require_relative '../../../spec_helper.rb'
+
+describe Salus::Scanners::Slither do
+  describe '#should_run?' do
+    it 'should return false in the absence of Solidity files' do
+      repo = Salus::Repo.new('spec/fixtures/blank_repository')
+      expect(repo.sol_file_present?).to be_falsey
+
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(false)
+    end
+
+    it 'should return false with .sol files but no package.json/config files' do
+      repo = Salus::Repo.new('spec/fixtures/slither/pure-solidity-good')
+      expect(repo.sol_file_present?).to be_truthy
+
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(false)
+    end
+
+    it 'should return false with .sol file, package.json but no config file' do
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity1')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(false)
+    end
+
+    it 'should return false with package.json, config file but no sol file' do
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity2')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(false)
+    end
+
+    it 'should return false with .sol file, truffle config file but no package.json' do
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity3')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(false)
+    end
+
+    it 'should return true with .sol file, package.json, and truffle config' do
+      # has package.json and truffle.js
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity_truffle_js')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(true)
+
+      # has package.json and truffle.ts
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity_truffle_ts')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(true)
+
+      # has package.json and truffle-config.js
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity_truffle_config_js')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(true)
+
+      # has package.json and truffle-config.ts
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity_truffle_config_ts')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner.should_run?).to eq(true)
+    end
+
+    it 'should return true with .sol file, package.json, and hardhat config' do
```

Comment:
Please split this example into multiple examples like this [comment](https://github.com/coinbase/salus/pull/623/files#r890308534).

Reply:
done